tests/transforms/test_clifford_t/test_solovay_kitaev.py

```diff
@@ -0,0 +1,206 @@
+
+    # Verify sequence check in check_nodes method
+    treeset4 = TreeSet(GateSet(), ["x", "z"], [])
+    seqs1 = GateSet([qml.PauliX(0)])
+    seqs2 = [GateSet([]), GateSet([qml.PauliX(0)]), GateSet([qml.PauliX(0), qml.PauliZ(0)])]
+    assert not treeset4.check_nodes(seqs1, seqs2)
+
+
+class TestSolovayKitaev:
+    """Test for Sololvay-Kitaev functionality"""
+
+    @pytest.mark.parametrize(
+        ("op"),
+        [
+            qml.U3(1.0, 2.0, 3.0, wires=[1]),
+            qml.U3(-1.0, 2.0, 3.0, wires=["b"]),
+            qml.U3(1.0, 2.0, -3.0, wires=[0]),
+        ],
+    )
+    def test_group_commutator_decompose(self, op):
+        """Test group commutator deocompose method for SU(2)"""
+
+        matrix = qml.matrix(op)
+        su2_matrix = matrix / qml.math.sqrt((1 + 0j) * qml.math.linalg.det(matrix))
+
+        v_hat, w_hat = _group_commutator_decompose(su2_matrix)
+        decomposed_matrix = (
+            v_hat
+            @ w_hat
+            @ qml.math.conj(qml.math.transpose(v_hat))
+            @ qml.math.conj(qml.math.transpose(w_hat))
+        )
+
+        assert qml.math.allclose(decomposed_matrix, su2_matrix)
+
+    @pytest.mark.parametrize(
+        ("op", "axis", "angle"),
+        [
+            (qml.Identity(wires=[0]), [0.0, 0.0, 1.0], 0.0),
+            (qml.S(wires=[0]), [0.0, 0.0, 1.0], math.pi / 2),
+            (qml.Hadamard(wires=["a"]), [1 / math.sqrt(2), 0.0, 1 / math.sqrt(2)], math.pi),
+        ],
+    )
+    def test_unitary_bloch(self, op, axis, angle):
+        """Test for finding bloch vector and angle for an SU(2)"""
+
+        matrix = qml.matrix(op)
+        su2_matrix = matrix / qml.math.sqrt((1 + 0j) * qml.math.linalg.det(matrix))
+
+        op_axis, op_angle = _unitary_bloch(su2_matrix)
+
+        assert qml.math.allclose(op_axis, axis) and qml.math.allclose(op_angle, angle)
+
+    def test_sk_approximate_set(self):
+        """Test for building approximate sets"""
+
+        approx_set1 = sk_approximate_set(depth=5)
+        approx_set2 = sk_approximate_set(basis_set=["t", "tdg", "h"], depth=5)
+
+        assert approx_set1 == approx_set2
+        assert len(approx_set1) < 263 and len(approx_set2) < 263  # 3 + ... + 3x3x3x3x3
+
+    @pytest.mark.parametrize(
+        ("op"),
+        [
+            qml.RX(math.pi / 42, wires=[1]),
+            qml.RZ(math.pi / 128, wires=[0]),
+            qml.RY(math.pi / 7, wires=["a"]),
+        ],
+    )
+    def test_solovay_kitaev(self, op):
+        """Test Solovay-Kitaev decomposition method"""
+
+        approximate_set = sk_approximate_set(
+            basis_set=[
+                "t",
+                "tdg",
+                "h",
+            ],
+            depth=10,
+        )
+        gates = sk_decomposition(op, depth=5, approximate_set=approximate_set)
+
+        matrix_sk = functools.reduce(lambda x, y: x @ y, map(qml.matrix, gates))
+        matrix_sk /= qml.math.sqrt((1 + 0j) * qml.math.linalg.det(matrix_sk))
```

Comment:
Is there some normalization happening somewhere? Or why do we need this line? :thinking:

Reply:
This is needed to find the SU(2) form of the unitary matrix U, which is essentially `SU2 = U * 1/sqrt(det(U))`, i.e., SU2 needs to have determinant as one.